src/dmd/cppmangle.d

```diff
@@ -1047,8 +1047,8 @@ public:
             case Tfloat80:               c = 'e';       break;
             case Tbool:                  c = 'b';       break;
             case Tchar:                  c = 'c';       break;
-            case Twchar:                 c = 't';       break;  // unsigned short (perhaps use 'Ds' ?
-            case Tdchar:                 c = 'w';       break;  // wchar_t (UTF-32) (perhaps use 'Di' ?
+            case Twchar:        p = 'D'; c = 's';       break;  // since C++11
+            case Tdchar:        p = 'D'; c = 'i';       break;  // since C++11
```

Comment:
The TypeBasic visitor guarded by Target.cppTypeMangle, right? Maybe is about time that we allow the user to control the c++ abi version than breaking backwards compatibility.